engine.go

```diff
@@ -661,12 +662,13 @@ func (e *RTCEngine) handleLeave(leave *livekit.LeaveRequest) {
 	if leave.GetCanReconnect() {
 		e.handleDisconnect(true)
 	} else {
+		reason := leave.GetReason()
 		e.log.Infow("server initiated leave",
-			"reason", leave.GetReason(),
+			"reason", reason,
 			"canReconnect", leave.GetCanReconnect(),
 		)
 		if e.OnDisconnected != nil {
-			e.OnDisconnected(LeaveRequested)
+			e.OnDisconnected(GetDisconnectionReason(reason))
 		}
```

Comment:
not about this PR, but we need to update this SDK to support LeaveRequest.Action at some point

Reply:
Will add a TODO here :+1: